scripts/treedb_vectordbbench_artifact.py

```
@@ -728,6 +747,434 @@ def file_identity(path: Path) -> dict[str, Any]:
+    try:
+        parsed = _dt.datetime.fromisoformat(value.replace("Z", "+00:00"))
+    except ValueError:
+        errors.append(f"{label} must be an RFC3339 timestamp")
+        return None
+    if parsed.tzinfo is None:
+        errors.append(f"{label} must include a timezone")
+        return None
+    return parsed.astimezone(_dt.timezone.utc)
+
+
+def validate_lifecycle_artifact(root: Path) -> dict[str, Any]:
```

Ruff (0.16.2)

[warning] Too many branches (101 > 12)

(PLR0912)